### PR TITLE
HARP-10769: Fix frame complete event

### PR DESCRIPTION
**Pull request by ninok**

### CODE REVIEW

File: @here/harp-mapview/lib/MapView.ts [hunk 1]
```diff
@@ -3503,15 +3503,16 @@ export class MapView extends THREE.EventDispatcher {
...
-            !this.isDynamicFrame &&
-            !this.textElementsRenderer.loading
+            !this.m_animatedExtrusionHandler.isAnimating
         ) {
-            this.m_firstFrameComplete = true;
-
-            if (gatherStatistics) {
-                stats.appResults.set("firstFrameComplete", frameStartTime);
+            if (this.m_firstFrameComplete) {
+                this.m_firstFrameComplete = true;
```
Comment (nzjony) [inline @here/harp-mapview/lib/MapView.ts:3512]: Not needed.

File: @here/harp-mapview/lib/MapView.ts [hunk 2]
```diff
@@ -3503,15 +3503,16 @@ export class MapView extends THREE.EventDispatcher {
...
             this.m_initialTextPlacementDone &&
-            !this.isDynamicFrame &&
-            !this.textElementsRenderer.loading
+            !this.m_animatedExtrusionHandler.isAnimating
         ) {
-            this.m_firstFrameComplete = true;
-
-            if (gatherStatistics) {
-                stats.appResults.set("firstFrameComplete", frameStartTime);
+            if (this.m_firstFrameComplete) {
```
Comment (nzjony) [inline @here/harp-mapview/lib/MapView.ts:3511]: IMO this should be renamed, as it is no longer the first frame

File: @here/harp-mapview/lib/MapView.ts [hunk 3]
```diff
@@ -3503,15 +3503,16 @@ export class MapView extends THREE.EventDispatcher {
         // running. The initial placement of text in this render call may have changed the loading
```
Comment (nzjony) [inline @here/harp-mapview/lib/MapView.ts:3503]: This comment should be fixed, it mentions above "first complete frame"

File: @here/harp-mapview/lib/MapView.ts (hunk 2 above)
Comment (ninok) [reply @here/harp-mapview/lib/MapView.ts:3511]: For the performance test we capture the time it took until the first frame is complete. I kept that functionality even though it's questionable why this is done inside MapView and not in the test.

File: @here/harp-mapview/lib/MapView.ts (hunk 3 above)
Comment (ninok) [reply @here/harp-mapview/lib/MapView.ts:3503]: done

File: @here/harp-mapview/lib/MapView.ts [hunk 4]
```diff
@@ -155,7 +155,10 @@ export enum MapViewEventNames {
     AfterRender = "didrender",
     /** Called after the first frame has been rendered. */
     FirstFrame = "first-render",
-    /** Called when the first view has all the necessary tiles loaded and rendered. */
+    /**
+     * Called when the rendered frame was complete, i.e. all the necessary tiles and resources
+     * where loaded and rendered.
```
Comment (nzjony) [inline @here/harp-mapview/lib/MapView.ts:160]:
```suggestion
     * are loaded and rendered.
```

File: @here/harp-mapview/lib/MapView.ts (hunk 1 above)
Comment (nzjony) [reply @here/harp-mapview/lib/MapView.ts:3515]: This is still missing, i.e. if m_firstFrameComplete is true, we don't need to set it true

Comment (ninok) [reply @here/harp-mapview/lib/MapView.ts:3485]: Oh, should be `if (this.m_firstFrameComplete === false)` instead. Fixed it

**Comment by ninok** [comment]
@heremaps-bot retest this please

**Pull request merged**